src/NosCore.Controllers/CommandPacketController.cs

```diff
@@ -198,16 +198,45 @@ public void Teleport(TeleportPacket teleportPacket)
         }
 
         [UsedImplicitly]
-        public void Gold(GoldCommandPacket goldPacket)
+        public void SetGold(SetGoldCommandPacket goldPacket)
         {
-            if (goldPacket.Gold + Session.Character.Gold > _worldConfiguration.MaxGoldAmount)
+            if (goldPacket.Name == Session.Character.Name || string.IsNullOrEmpty(goldPacket.Name))
             {
-                Session.SendPacket(Session.Character.GenerateSay(goldPacket.Help(), SayColorType.Yellow));
+                Session.Character.SetGold(goldPacket.Gold);
```

Comment:
check goldPacket.Gold + Gold doesnt overflow

Reply:
you also need a test here for your own session or even better just use the api for yourself too.